agent/xds/listeners_ingress.go

```diff
@@ -284,33 +360,68 @@ func makeSDSOverrideFilterChains(cfgSnap *proxycfg.ConfigSnapshot,
 	return chains, nil
 }
 
-func makeCommonTLSContextFromSDS(sdsCfg structs.GatewayTLSSDSConfig) *envoy_tls_v3.CommonTlsContext {
+var envoyTLSVersions = map[types.TLSVersion]envoy_tls_v3.TlsParameters_TlsProtocol{
+	types.TLSVersionAuto: envoy_tls_v3.TlsParameters_TLS_AUTO,
+	types.TLSv1_0:        envoy_tls_v3.TlsParameters_TLSv1_0,
+	types.TLSv1_1:        envoy_tls_v3.TlsParameters_TLSv1_1,
+	types.TLSv1_2:        envoy_tls_v3.TlsParameters_TLSv1_2,
+	types.TLSv1_3:        envoy_tls_v3.TlsParameters_TLSv1_3,
+}
+
+func makeTLSParametersFromGatewayTLSConfig(tlsCfg structs.GatewayTLSConfig) *envoy_tls_v3.TlsParameters {
+	tlsParams := envoy_tls_v3.TlsParameters{}
+
+	if tlsCfg.TLSMinVersion != types.TLSVersionUnspecified {
+		tlsParams.TlsMinimumProtocolVersion = envoyTLSVersions[tlsCfg.TLSMinVersion]
```

Comment:
Is there a chance that these can ever be set to 0 values of `envoy_tls_v3.TlsParameters_TlsProtocol`? (In the case that the key doesn't exist in `envoyTLSVersions` for instance)

Reply:
Should probably explicitly handle not finding the string in the map too.

Reply:
It shouldn't be possible at this point, as these values should have been validated in when the config entry was applied and there's no way to return an error to the user from within here - implemented checks but is there a good way to log something if this value is somehow invalid rather than silently skipping config?